OpenSim/Framework/Communications/UserProfileManager.cs

```diff
@@ -240,6 +240,12 @@ public UserProfileData GetUserProfile(UUID uuid, bool forceRefresh)
                 profile = TryGetUserProfile(uuid, false);
                 if (profile != null)
                 {
+                    if (profile.CurrentAgent == null)
```

Comment:
I think you might need to have `&& !(m_localUser.ContainsKey(uuid) || m_tempDataByUUID.ContainsKey(uuid))` in the conditional or you wind up violating the comment below that says "Never force refresh of local profiles or temp profiles."